.pre-commit-config.yaml

```diff
@@ -0,0 +1,64 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+    # General
+    -   id: check-merge-conflict
+        fail_fast: true
+    -   id: check-case-conflict
+        fail_fast: true
+    -   id: check-added-large-files
+    -   id: check-symlinks
+    -   id: destroyed-symlinks
+    # Syntax
+    -   id: check-json
+    -   id: check-yaml
+    # Whitespace
+    -   id: mixed-line-ending
+    -   id: trailing-whitespace
+
+-   repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+    # Python
+    -   id: python-use-type-annotations
+    -   id: python-no-log-warn
+
+    # Documentation
+    -   id: rst-backticks
+    -   id: rst-directive-colons
+    -   id: rst-inline-touching-normal
+
+-   repo: https://github.com/abravalheri/validate-pyproject
+    rev: v0.10.1
+    hooks:
+    - id: validate-pyproject
+
+-   repo: https://github.com/asottile/setup-cfg-fmt
+    rev: v2.2.0
+    hooks:
+    -   id: setup-cfg-fmt
+
+-   repo: https://github.com/charliermarsh/ruff-pre-commit
```

Comment:
Would this mean we might want to collectively agree to switch to ruff as our primary linter?

Reply:
I don't know if `ruff` supports all options of `pylint` or not. If you want to read up about it: https://github.com/charliermarsh/ruff
You can enable a bunch of things I did not, I just made this rather inoffensive. I have no issue with switching, but I think that's a different PR / discussion.